Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -15,8 +15,83 @@
+/// for an offline enabled web map. The view:
+///
+/// - Displays a list of map areas.
+/// - Shows download progress and status for map areas.
+/// - Opens a map area for viewing when selected.
+/// - Provides options to view details about downloaded map areas.
+/// - Supports removing downloaded offline map areas files from the device.
+///
+/// For preplanned workflows, the view:
+///
+/// - Displays a list of available preplanned map areas from an offline-enabled
+/// web map that contains preplanned map areas when the network is connected.
+/// - Downloads preplanned map areas in the list.
+/// - Displays a list of downloaded preplanned map areas on the device
+/// when the network is disconnected.
+///
+/// For on-demand workflows, the view:
+///
+/// - Allows users to add and download on-demand map areas to the device by
+/// specifying an area of interest and level of detail.
+/// - Displays a list of on-demand map areas available on the device that are
+/// tied to a specific web map.
+///
+/// **Behavior**
+///
+/// Depending on whether the `OfflineMapAreasView` is presented modally or
+/// embedded in a navigation stack, you can use the ``doneButton(_:)`` modifier
+/// to set the visibility of the done button. The done button dismisses the
+/// view. The button hides when its visibility is set to `.hidden`. The default
+/// visibility is `.automatic`.
+///
+/// The view can be initialized with a web map or an offline map info.
+/// Therefore, the view can be used either when the device is connected to
+/// or disconnected from the network. In other words, the view can be used
+/// in the following situations:
+///
+/// - When the device is connected to the network…
+///     - The view displays preplanned map areas from a web map that are
+///     available for download.
+///     - When the web map doesn't contain preplanned map areas, users can add
+///     and download on-demand map areas.
+///     - Use ``init(onlineMap:selection:)`` to initialize the view.
+/// - When the device is disconnected from the network…
+///     - The view displays only downloaded map areas by retrieving
+///     offline map info from the device.
+///     - Use ``init(offlineMapInfo:selection:)`` to initialize the view.
+///
+/// When the network connection is available, the component will display a list
+/// of preplanned map areas configured for the web map. If the web map doesn't
+/// contain any preplanned map areas, users can create a map area on-demand by
+/// specifying a geographic area and level of detail. When a network connection
+/// is unavailable, the component will only display map areas that have already
+/// been downloaded to the device.
+///
+/// If the network connectivity changes while the view is presented, the view
+/// will not automatically refresh the list of map areas. The user can
+/// pull-to-refresh to reload the list of map areas. Depending on the network
+/// connectivity, the view will display the appropriate content.
```

Comment:
Divesh suggested to remove this paragraph as it might be more of an implementation detail than a useful documentation. However, I'm debating myself over whether or not we should mention that the view won't switch between preplanned vs on-demand automatically.

So I leave this in for now and welcome some thoughts and comments.

Reply:
I can't think of a good reason to keep it, and since Divesh said to remove it, that would also be my recommendation in this case.